tests/app_test.rb

```diff
@@ -0,0 +1,63 @@
+ENV['RACK_ENV'] = 'test'
+
+require_relative '../app'
+require 'test/unit'
+require 'rack/test'
+require 'pry'
+
+
+configure do
+  DataMapper::setup(:default, "sqlite3:test.db")
+  DataMapper.finalize
+  DataMapper.auto_migrate!
+end
+
+class ApplicationControllerTest < Test::Unit::TestCase
+  include Rack::Test::Methods
+
+  def app
+    @app ||= ApplicationController.new
+  end
+
+  def test_home_page
+    get '/'
+    last_response.body
+    assert last_response.ok?
+  end
+
+  def test_show_form
+    get '/show'
+    assert last_response.ok?
+  end
+
+  def test_view_details
```

Comment:
Here also, we want to first call the `POST details/new` endpoint with some data and then check for some text in index.